src/types.jl

```diff
@@ -97,13 +52,416 @@ for Model in [:NeuralNetworkClassifier, :ImageClassifier]
+    - `Flux.binary_focal_loss`: Binary version of the above
+- `epochs::Int=10`: The number of epochs to train for. Typically, one epoch represents one pass through the entirety of the training dataset.
+- `batch_size::Int=1`: The batch size to be used for training. The batch size represents the number of samples per update of the networks weights. Typcally, batch size should be somewhere between 8 and 512. Smaller batch sizes lead to noisier training loss curves, while larger batch sizes lead towards smoother training loss curves. In general, it is a good idea to pick one fairly large batch size (e.g. 32, 64, 128), and stick with it, and only tune the learning rate. In most literature, batch size is set in powers of twos, but this is fairly arbitrary.
+- `lambda::Float64=0`: The stregth of the regularization used during training. Can be any value  in the range `[0, ∞)`.
+- `alpha::Float64=0`: The L2/L1 mix of regularization, in the range `[0, 1]`. A value of 0 represents L2 regularization, and a value of 1 represents L1 regularization.
+- `rng::Union{AbstractRNG, Int64}`: The random number generator/seed used during training.
+- `optimizer_changes_trigger_retraining::Bool=false`: Defines what happens when fitting a machine if the associated optimiser has changed. If true, the associated machine will retrain from scratch on `fit`, otherwise it will not.
+- `acceleration::AbstractResource=CPU1()`: Defines on what hardware training is done. For Training on GPU, use `CudaLibs()`, otherwise defaults to `CPU`()`.
+- `finaliser=Flux.softmax`: The final activation function of the neural network. Defaults to `Flux.softmax`. For a classification task, `softmax` is used for multiclass, single label regression, `sigmoid` is used for either binary classification or multi label classification (when there are multiple possible labels for a given sample).
+
+
+# Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given new
+  features `Xnew` having the same Scitype as `X` above. Predictions are
+  probabilistic.
+- `predict_mode(mach, Xnew)`: Return the modes of the probabilistic predictions
+  returned above.
+
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+- `chain`: The trained "chain", or series of layers, functions, and activations which make up the neural network.
+
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `training_losses`: The history of training losses, a vector containing the history of all the losses during training. The first element of the vector is the initial penalized loss. After the first element, the nth element corresponds to the loss of epoch n-1.
+
+# Examples
+
+In this example we build a classification model using the Iris dataset.
+```julia
+using MLJ
+using Flux
+import RDatasets
+
+using Random
+Random.seed!(123)
+
+MLJ.color_off()
+
+using Plots
+pyplot(size=(600, 300*(sqrt(5)-1)));
```

Comment:
Let's use the fallback GR plotting to avoid PyPlot as an extra dependency. This is more reliable. And we can postpone the `using Plots` call until later (actually you have it already again below) to save the extra compilation until needed. 

```suggestion
```